tezos/new_context/src/hash/mod.rs

```diff
@@ -90,23 +90,23 @@ fn index(depth: u32, name: &str) -> u32 {
 // IMPORTANT: entries must be sorted in lexicographic order of the name
 // Because we use `OrdMap`, this holds true when we iterate the items, but this is
 // something to keep in mind if the representation of `Tree` changes.
-fn partition_entries<'a>(depth: u32, entries: &[(&Rc<String>, &'a Node)]) -> Result<Inode<'a>, HashingError> {
+fn partition_entries<'a>(
+    depth: u32,
+    entries: &[(&KeyFragment, &'a Node)],
+) -> Result<Inode<'a>, HashingError> {
     if entries.is_empty() {
         Ok(Inode::Empty)
     } else if entries.len() <= 32 {
         Ok(Inode::Value(
-            entries
-                .iter()
-                .map(|(s, n)| (Rc::clone(*s), *n))
-                .collect(),
+            entries.iter().map(|(s, n)| ((*s).clone(), *n)).collect(),
```

Comment:
@sebastiencs isn't `Rc::clone` the idiomatic way to handle these cases where you clone an `Rc` ?

Reply:
Oh, duh, sorry, just noticed that it is now a `KeyFragment`, ignore me.

Reply:
It is, but here we're cloning a `KeyFragment`